examples/guide/references.rs

```diff
@@ -0,0 +1,102 @@
+#![cfg_attr(verus_keep_ghost, verifier::exec_allows_no_decreases_clause)]
+
+#[allow(unused_imports)]
+use builtin::*;
+#[allow(unused_imports)]
+use builtin_macros::*;
+use vstd::prelude::*;
+
+verus! {
+
+
+fn immut()
+{
+// ANCHOR: immut
+let x: u32 = 0;
+let immut_ref_x = &x;
+assert(x == 0);
+assert(*immut_ref_x == 0);
+// ANCHOR_END: immut
+}
+
+// ANCHOR: mut
+fn mutable_example()
+{
+    let y: u32 = 1;
+    assert(y == 1);
+    modify_y(&mut y);
+    assert(y == 2);
+}
+
+fn modify_y(a: &mut u32) 
+    ensures *a == 2
+{
+    *a = 2;
+}
+// ANCHOR_END: mut
+
+// ANCHOR: requires
+fn function_requires_ensures()
+{
+    let mut z: u32 = 0;
+    check_and_inc(&mut z);
+}
+
+fn check_and_inc(a: &mut u32) 
+    requires *old(a) == 0
+    ensures *a == 1
+{
+    *a = *a + 1;
+}
+// ANCHOR_END: requires
+
+// ANCHOR: asserts
+fn asserts() 
+{
+    let mut x: u32 = 0;
+    check_and_assert(&mut x);
+}
+
+fn check_and_assert(a: &mut u32)
+    requires *old(a) == 0
+{
+    assert(*old(a) == 0);
+    *a = *a + 1;
+    assert(*a == 1);
+    *a = *a + 1;
+    assert(*a == 2);
```

Comment:
Maybe we could add another copy of `assert(*old(a) == 0);` in here to emphasize that `old` refers to the value at the very beginning of the function, not to the value right before the last update.